src/index.js

```diff
@@ -18,13 +19,15 @@ ReactDOM.render(
   <React.StrictMode>
     <Router>
       <ThemeProvider>
+      <AuthProvider>
         <VideoProvider>
           <FilterProvider>
             <ServerProvider>
               <App />
             </ServerProvider>
           </FilterProvider>
         </VideoProvider>
+      </AuthProvider>
       </ThemeProvider>
     </Router>
```

Comment:
Instead of writing all the providers on the App file itself, you can move them in one wrapper and use it here for the leaner structure.